src/main/java/io/projectriff/invoker/server/GrpcServerAdapter.java

```diff
@@ -138,6 +146,47 @@ private OutputSignal fromSpringMessage(Tuple2<Integer, Message<byte[]>> out) {
                 ;
     }
 
+    // Used to transform the publisher chain into one that doesn't forward cancel() calls once it has complete()d.
+    private Function<? super Publisher<Tuple2<Integer, Message<byte[]>>>, ? extends Publisher<Tuple2<Integer, Message<byte[]>>>> ignoreCancelsAfterComplete() {
+        return Operators.lift((f, actual) ->
+                new CoreSubscriber<>() {
```

Comment:
FTR this can be optimized by implementing both `CoreSubscriber` and `Subscription`, something like:
```java
class NoCancelAfterCompleteSubscriber<T> implements CoreSubscriber<T>, Subscription {

    @Override
    public void onSubscribe(s) { actual.onSubscribe(this); }
}

return new NoCancelAfterCompleteSubscriber<>();
```

although this is a very specific optimization can may be ignored if the performance of this part is not critical